lib/Config.ts

```diff
@@ -110,6 +116,16 @@ class Config {
           this.updateDefaultPaths(props.xudir);
         }
 
+        if (props.network !== undefined && (!args || !args.network)) {
+          // first check that network is a valid value
+          if (typeof props.network !== 'string' || !Object.keys(Network).some((key: string) => key.toLowerCase() === props.network)) {
```

Comment:
Why not `!Object.keys(Network).includes(`?

Reply:
Good question, I suppose I could add a comment because it's a bit complicated (although not very important).

The `Network` enum is basically an object with keys for all the enum values (`mainnet, `testnet`, etc..) as well as the enum names (`MainNet`, `TestNet`, etc...).  I only want to check for a match against the lower case `enum` values, and this was the only way I could figure out how to do it. `includes` would match with a value of `network = "MainNet"` which I'd want to avoid.

Reply:
If you want to compare against the values of the `enum` (which are all lower case) why are you using `Object.keys` then?

`!Object.values(Network).includes(props.network)` would do the exact same thing or am I missing something? @sangaman

Reply:
Same thing. In an enum, the set of keys is the same as the set of values. Took me a while to figure this out.

Reply:
I am not sure I get what you mean:

`Object.keys(Network)` is equal: `[ 'MainNet', 'TestNet', 'SimNet', 'RegTest' ]` but
`Object.values(Network)` is: `[ 'mainnet', 'testnet', 'simnet', 'regtest' ]`

So `Object.keys(Network).some((key: string) => key.toLowerCase() === props.network)` should always be equal to `Object.values(Network).includes(props.network)`? @sangaman

Reply:
Wow well I tested this out and I was totally wrong, either this has changed in TypeScript from older versions or I just misinterpreted it off of answers like this one: https://stackoverflow.com/a/39439520/9137842

Will change this to use `values` and `includes`. Thanks for the lesson.